### PR TITLE
Releasecandidate/omni

### PR DESCRIPTION
**Pull request by wagenrace**
Thank you for your contribution to the **CytoSmart Open API** package. 
Before submitting/accepting this Pull Request, please make sure:

## Contributor of code
- [x] Functions are named based on [Python naming conventions](https://visualgit.readthedocs.io/en/latest/pages/naming_convention.html)
- [x] Explain how to run the code `(docstring)`
- [x] Function is tested
- [x] All needed package are in [requirements.txt](requirements.txt)
- [x] Version number is updated
- [x] `HISTORY.rst` is updated
- [x] New feature is added to `README.md` subsection `Features`

## Reviewer
- [ ] Functions are named based on [Python naming conventions](https://visualgit.readthedocs.io/en/latest/pages/naming_convention.html)
- [ ] The code is executed on another machine
- [ ] Function is tested
- [ ] Installing just the [requirements.txt](requirements.txt) is enough to run the code
- [ ] Version number is updated
- [ ] `HISTORY.rst` is updated
- [ ] New feature is added to `README.md` subsection `Features`

### CODE REVIEW

File: CytoSmartOpenAPI/listener.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+#%%
+import json
+from threading import Thread
+import time
+from typing import Dict, Callable
+
+from .cytosmart_device import CytoSmartDevice
+
+
+class Listener(Thread):
```
Comment (schneiderjan) [inline CytoSmartOpenAPI/listener.py:10]: Add doc string across the class.

File: CytoSmartOpenAPI/main.py [hunk 2]
```diff
@@ -14,21 +14,17 @@
...
-        try:
-            # Try to make a connection with the app
-            self.ws = create_connection("ws://localhost:3333/cytosmartservice")
-        except:
-            # If that fails, start the app first and make the connection again
-            self.__start_lux_app()
-            self.ws = create_connection("ws://localhost:3333/cytosmartservice")
-        self.ws_listener = Listener(self.ws)
+        if warranty:
+            raise ValueError("The warranty MOST be set to False")
```
Comment (schneiderjan) [inline CytoSmartOpenAPI/main.py:25]: must

File: CytoSmartOpenAPI/main.py [hunk 3]
```diff
@@ -49,27 +45,81 @@ def __init__(self, number_of_devices: int = 1) -> None:
             except:
                 pass
 
+    def __connect_with_service(self):
+        """
+        Try to connect with CytoSmartService websocket.
+        If that does not work, start a instants of the service
```
Comment (schneiderjan) [inline CytoSmartOpenAPI/main.py:51]: instance

File: CytoSmartOpenAPI/main.py [hunk 4]
```diff
@@ -213,7 +259,35 @@ def set_camera_settings(
...
+
+    def set_flash_duration(self, serial_number: str, duration: int) -> None:
+        """
+        Setting the time in μs the led is on during the taking of a picture.
+
+        Args:
+        serial_number: (str) the serial number of device you want to connect
+            duration (int): μs between 40 and 250
+        """
+        assert 40 <= duration and duration <= 250
```
Comment (schneiderjan) [inline CytoSmartOpenAPI/main.py:284]: If this is user facing you can add also a message, `assert condition, "your message"`

File: README.md [hunk 5]
```diff
@@ -1,40 +1,57 @@
...
-Make sure you have or had a [CytoSmart application installed](http://download.cytosmart.com/).
-Recommend is the cell counter given it do not restart itself after closing.
+## Installation
+
+To install this package follow the following steps:
+
+
+### Step 1: Drivers
+
+Make sure you had a [CytoSmart application installed](http://download.cytosmart.com/).
```
Comment (schneiderjan) [inline README.md:27]: have the

File: README.md [hunk 6]
```diff
@@ -1,40 +1,57 @@
...
+ 
+This is because our devices are made to handle the normal usage ([the GUI app](http://download.cytosmart.com/)).
+It also includes additionally bought warranty.
+Only if your additionally bought warranty implicitly includes the open API usage you will have warranty.
 
-Make sure you have or had a [CytoSmart application installed](http://download.cytosmart.com/).
-Recommend is the cell counter given it do not restart itself after closing.
+## Installation
+
+To install this package follow the following steps:
```
Comment (schneiderjan) [inline README.md:22]: ...package follow these steps:

**Review by Bram-van-der-Velden — APPROVED** [review]

**Pull request merged**